element-hq/element-x-android
# Improve room setting edition

## Type of change


## Content



May be reviewed commit per commit, first commits are cleanup.

Ensure that when editing room setting, the actual `m.room.name` value is used to fill the form, and not the computed name. Closes #2844, but there is an issue SDK side, rawName contains the displayName. It's discussed internally:

<img width="600" alt="image" src="https://github.com/element-hq/element-x-android/assets/3940906/d664291b-aabe-45e9-ac79-ef87e48b9834">


Also make the room details (topic/name/avatar) live in this screen, in case it's changes by someone else, or by another session.

## Motivation and context



## Screenshots / GIFs



## Tests



- Open room setting with a no-name room and observe that the room name field is empty. (Note: see the SDK limitation above)
- Make some change and ensure that the 'Save' action works as expected and is enabled when expected.

## Tested devices

- OS version(s):

## Checklist
